test/common/filter/tcp_proxy_test.cc

```diff
@@ -824,7 +791,8 @@ TEST_F(TcpProxyTest, AccessLogUpstreamLocalAddress) {
 TEST_F(TcpProxyTest, AccessLogDownstreamAddress) {
   Network::Address::InstanceConstSharedPtr downstream_address =
       Network::Utility::resolveUrl("tcp://1.1.1.1:40000");
-  ON_CALL(filter_callbacks_.connection_, remoteAddress()).WillByDefault(Return(downstream_address));
+  ON_CALL(filter_callbacks_.connection_, remoteAddress())
+      .WillByDefault(ReturnRef(downstream_address));
```

Comment:
Ditto here.
And would it make sense to have this be the default behavior or would that cause problems for other tests?